Factorial/src/Factorial.java

```diff
@@ -0,0 +1,80 @@
+import org.junit.jupiter.api.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+import static org.junit.jupiter.api.Assertions.*;
+
+/* Reads an integer and prints n! = 1 * 2 * ... * n to standard output
+   For eg: 0! = 1, 1! = 1, 2! = 2 * 1 = 2, 3! = 3 * 2 * 1 = 6 ..  so on
+   Remarks
+    - There would be an overflow for long if n > 20
+    (Factorial for 21 will give you -> -4249290049419214848 which is incorrect)
+*/
+public class Factorial {
+    // Return n!
+    public static long factorialUsingRecursion(long n) {
+        if (n < 0) {
+            throw new RuntimeException("Underflow error in factorial");
```

Comment:
Should be IllegalArgumentException

Reply:
IllegalArgumentException extends the RuntimeException 
Ideally the greater than 20 value is not illegal? Just that the long data type will not be able to hold that size after computation right?

Hence did not add IllegalArgumentException then. But for the code it makes more sense to have this. Hence updating it.